src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -50,13 +50,20 @@ internal struct Builder
             /// </remarks>
             internal bool InterlockedInitialize(TypeWithAnnotations type)
             {
-                if ((object)_defaultType != null)
+                if (!IsDefault)
                 {
                     return false;
                 }
                 _nullableAnnotation = type.NullableAnnotation;
```

Comment:
This needs to be set with `Interlocked.Exchange`

Reply:
There's no version that works with a `byte`. I'll change the enum to use `int` then...